src/kbmod/search/KBMOSearch.cpp

```diff
@@ -317,7 +317,7 @@ std::vector<RawImage> KBMOSearch::meanScienceStamps(const std::vector<Trajectory
 
 #pragma omp parallel for
     for (int s = 0; s < num_results; ++s) {
-        results[s] = meanScienceStamp(t_array[s], radius, true);
+        results[s] = meanScienceStamp(t_array[s], radius, false);
```

Comment:
ok so just to repeat back the logic of this fix to make sure we're on the same page, the previous change made it so that median stamps no longer only use GoodIdx, so they need to not use all stamps (hence this and the above change to false).

Reply:
Correct. Before 175 meanStamps contained the following loop:
`for (int i = 0; i < goodIdx[s].size(); ++i) {`
`if (goodIdx[s][i] == 1) {`
`pixelPos pos = getTrajPos(t, i);`
`stamps.push_back(imgs[i].getScience().createStamp(pos.x, pos.y, radius, false, true));`
`}`
`}`
So each time step was checked against `goodIdx` before pushing it into the array of stamps. And the mean/median was built on only those images inserted into the array.

This pruning logic was moved to `scienceStamps ` (line 254), but only applies if `all_stamps` is false. Otherwise, as the name implies, it uses all the stamps for the mean / median. I believe the error was passing true to `meanScienceStamp` which forces it to use all of the indices.